mobile/library/common/internal_engine.cc

```diff
@@ -294,6 +295,15 @@ envoy_status_t InternalEngine::setPreferredNetwork(NetworkType network) {
         connectivity_manager_->dnsCache()->setIpVersionToRemove(absl::nullopt);
       }
     }
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.reset_brokenness_on_nework_change")) {
+      Http::HttpServerPropertiesCacheManager& cache_manager =
+          server_->httpServerPropertiesCacheManager();
+
+      Http::HttpServerPropertiesCacheManager::CacheFn clear_brokenness =
+          [](Http::HttpServerPropertiesCache& cache) { cache.resetBrokenness(); };
+      cache_manager.forEachThreadLocalCache(clear_brokenness);
+    }
```

Comment:
I'm guessing there is no remotely easy way to test this, is there?

Reply:
not sure but we could possibly test this using something similar to https://github.com/envoyproxy/envoy/blob/0e03184359729cf95d8ed56a2b82ab93dde2d6bf/mobile/test/common/network/connectivity_manager_test.cc#L38

Reply:
I don't think there's really a good way to test it today - I can add a test in connectivity_manager_test.cc but it basically verifies it in isolation which has minimal value.

This is largely why I kicked off https://github.com/envoyproxy/envoy/pull/35551 - I think we need proper tests for more of the QUIC work.  I thought of doing them in C++ since that'd be faster for me but I think there's more value in having them for Java so I'm working on test tech debt in parallel.

Reply:
i think the PR is good as is